src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -2381,7 +2381,14 @@ var projectUniqueNamesForBuildIntToUpdate
                 }
                 else if (action.NuGetProjectActionType == NuGetProjectActionType.Install)
                 {
-                    PackageSpecOperations.AddOrUpdateDependency(updatedPackageSpec, action.PackageIdentity);
+                    if (updatedPackageSpec.RestoreMetadata.ProjectStyle == ProjectStyle.ProjectJson)
```

Comment:
Would it make more sense if you check for `PackageReference` and pass tfms. And else case use the old call. That will keep the existing behavior and only change PR